src/core/libmaven/PeakDetector.cpp

```diff
@@ -82,8 +83,16 @@ vector<EIC*> PeakDetector::pullEICs(mzSlice* slice,
                         EIC::SmootherType smootherType =
                                 (EIC::SmootherType) smoothingAlgorithm;
                         e->setSmootherType(smootherType);
-                        e->setBaselineSmoothingWindow(baseline_smoothingWindow);
-                        e->setBaselineDropTopX(baseline_dropTopX);
+
+                        // set appropriate baseline parameters
+                        e->setBaselineMode(baselineMode);
+                        if (baselineMode == EIC::BaselineMode::AsLSSmoothing) {
+                            e->setAsLSSmoothness(firstBaselineParameter);
+                            e->setAsLSAsymmetry(secondBaselineParameter);
+                        } else {
+                            e->setBaselineSmoothingWindow(firstBaselineParameter);
+                            e->setBaselineDropTopX(secondBaselineParameter);
```

Comment:
The same set of logic seems to be replicated else where. I believe, if I have not missed anything, we can get rid of this logic elsewhere also. Correct me if I am wrong

Reply:
I think I will have to try out these snippets before I can comment. While what you suggest reduces code quantity (when setting parameters each time like above), the current method makes more sense semantically - as in what would we do if the baseline parameter for a new algorithm takes in three arguments. I hope we do not need another such baseline implementation, but pointing this out just for consideration.

Reply:
> what would we do if the baseline parameter for a new algorithm takes in three arguments. I hope we do not need another such baseline implementation, but pointing this out just for consideration.

I agree. Just because these 2 algorithms happened to have 2 parameters, does not mean we should take advantage of it. 

Considering we're going to move towards metabolite specific settings soon, it might be a good idea to think of structs to bundle different sets of parameters together? All EIC settings like smoothing algo, window, baseline, baselineDiff etc can be stored in one struct and passed to this function. What do you guys think?

Reply:
Using a struct for passing a bunch of parameters around is not a bad idea. But we should carefully outline and define what these structures will be. Otherwise we might end up with another `MavenParameters` sort of thing. `MavenParameters` is not so bad though, and for now an instance of it can be passed to `pullEICs` to reduce a fair amount of arguments.

Reply:
I'm suggesting a struct for EIC parameters within MavenParameters. Instead of sending the full MavenParameters object to functions, we can pass that one struct.